packages/chat-core-aws-connect/src/infra/ChatCoreAwsConnectImpl.ts

```diff
@@ -130,4 +130,14 @@ export class ChatCoreAwsConnectImpl implements ChatCoreAwsConnect {
   getSession(): connect.ActiveChatSession | undefined {
     return this.session;
   }
+
+  resetSession(): void {
+    if (this.session === undefined) {
+      return;
+    }
+
+    const customerSession = this.session as connect.ActiveCustomerChatSession;
```

Comment:
can we update session type in L15 instead of doing type assertion? when setting `this.session`, we can do a type check to ensure it's a customer chat session first.

Reply:
Good idea! Added `ChatSession.ts` with the typeguard, then used that when initializing.